pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -770,7 +770,20 @@ func (m *kubeGenericRuntimeManager) killContainer(ctx context.Context, pod *v1.P
 	klog.V(2).InfoS("Killing container with a grace period", "pod", klog.KObj(pod), "podUID", pod.UID,
 		"containerName", containerName, "containerID", containerID.String(), "gracePeriod", gracePeriod)
 
-	err := m.runtimeService.StopContainer(ctx, containerID.ID, gracePeriod)
+	var err error
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		err = m.runtimeService.StopContainer(ctx, containerID.ID, gracePeriod)
+
+	}()
+	select {
+	case <-ctx.Done():
+		err = errors.New("container receives the ctx.Done signal and cancels the deletion")
```

Comment:
How about:
```
StopContainer request cancelled by caller
```
or something. My fundamental problem with the error is that technically here it's the this process that caught the ctx.Done, not the container

Reply:
> or something. My fundamental problem with the error is that technically here it's the this process that caught the ctx.Done, not the container

Yes, you're right, I'll change it here